org.eclipse.lemminx/src/main/resources/org/eclipse/lemminx/services/snippets/new-xml-snippets.json

```diff
@@ -0,0 +1,97 @@
+{
+  "New XML with SYSTEM DOCTYPE": {
+    "prefix": [
+      "<!DOCTYPE"
+    ],
+    "suffix": ">",
+    "body": [
+      "<!DOCTYPE ${1:root-element} SYSTEM \"${2:file}.dtd\" >",
+      "<${1:root-element}>${0}",
+      "</${1:root-element}>"
+    ],
+    "description": "New XML with SYSTEM DOCTYPE"
+  },
+  "New XML with PUBLIC DOCTYPE": {
+    "prefix": [
+      "<!DOCTYPE"
+    ],
+    "suffix": ">",
+    "body": [
+      "<!DOCTYPE ${1:root-element} PUBLIC \"${2:public-id}\" \"${3:file}.dtd\" >",
+      "<${1:root-element}>${0}",
+      "</${1:root-element}>"
+    ],
+    "description": "New XML with PUBLIC DOCTYPE"
+  },
+  "New XML with SYSTEM DOCTYPE with subset": {
+    "prefix": [
+      "<!DOCTYPE"
+    ],
+    "suffix": ">",
+    "body": [
+      "<!DOCTYPE ${1:root-element} SYSTEM \"${2:file}.dtd\" [",
+      "${0}",
+      "]>",
+      "<${1:root-element}>${0}",
+      "</${1:root-element}>"
+    ],
+    "description": "New XML with SYSTEM DOCTYPE with subset"
+  },
+  "New XML with PUBLIC DOCTYPE with subset": {
+    "prefix": [
+      "<!DOCTYPE"
+    ],
+    "suffix": ">",
+    "body": [
+      "<!DOCTYPE ${1:root-element} PUBLIC \"${2:public-id}\" \"${3:file}.dtd\" [",
+      "${0}",
+      "]>",
+      "<${1:root-element}>${0}",
+      "</${1:root-element}>"
+    ],
+    "description": "New XML with PUBLIC DOCTYPE with subset"
+  },
+  "New XML with DOCTYPE with subset": {
+    "prefix": [
+      "<!DOCTYPE"
+    ],
+    "suffix": ">",
+    "body": [
+      "<!DOCTYPE ${1:root-element} [",
+      "\t<!ELEMENT ${1:root-element} (${2:#PCDATA}) >",

```

Comment:
Remove extra space before the closing `>`

Reply:
fixed